packages/blade/src/components/Input/TextArea/TextArea.tsx

```diff
@@ -70,9 +70,11 @@ const TextArea = ({
 }: TextAreaProps): React.ReactElement => {
   const inputRef = React.useRef<HTMLInputElement | TextInputReactNative>(null);
 
-  const [shouldShowClearButton, setShouldShowClearButton] = React.useState<boolean>(
-    Boolean(showClearButton && (value?.length || defaultValue?.length)),
-  );
+  const [shouldShowClearButton, setShouldShowClearButton] = React.useState(false);
+
+  React.useEffect(() => {
```

Comment:
Could've just put this on the useState() but IMO it's better to keep this on useEffect to avoid any stale state. 

If we don't use useEffect, The input won't rerender and it will behave strangely, this will happen if while typing someone tries to change the showClearButton. 

Can be shown in storybook: 

https://user-images.githubusercontent.com/35374649/210552166-919964d6-e94a-42a8-b350-f39b2f0ac644.mov